vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -917,8 +918,17 @@ def get_rope(
         return _ROPE_DICT[key]
 
     if rope_scaling is None:
-        rotary_emb = RotaryEmbedding(head_size, rotary_dim, max_position, base,
-                                     is_neox_style, dtype)
+        if current_platform.is_hpu():
+            from vllm_hpu_extension.rotary_embed import HpuRotaryEmbedding
+            rotary_emb = HpuRotaryEmbedding(head_size,
+                                            rotary_dim,
+                                            max_position,
+                                            base,
+                                            is_neox_style,
+                                            RoPEFallback=RotaryEmbedding)
```

Comment:
QQ: Can we implement this in `forward_hpu` of `RotaryEmbedding`?

Reply:
sure, just did 😺 The changes in RoPE are now limited to forward_hpu